modules/unsupported/geopkg/src/main/java/org/geotools/geopkg/geom/GeoPkgGeomWriter.java

```diff
@@ -28,13 +28,23 @@ public void setWriteEnvelope(boolean writeEnvelope) {
     }
     
     protected Configuration config;
+    protected int dim;
         
     public GeoPkgGeomWriter() {
-        config = new Configuration();
+        this(2, new Configuration());
     }
-
+    
+    public GeoPkgGeomWriter(int dim) {
+        this(dim, new Configuration());
+    }
+    
     public GeoPkgGeomWriter(Configuration config) {
+        this(2, config);
+    }
+
+    public GeoPkgGeomWriter(int dim, Configuration config) {
         this.config = config;
+        this.dim = dim;
```

Comment:
Hmm only the one without arguments and with all arguments are currently being used, but I wanted to allow some flexibility for the possible future with respect to the default constructor arguments (it isn't always all or nothing).